### ISSUE TITLE
[Suggestion] Weapon building changes

### ISSUE THREAD

**Issue by ultrak2k** [tsunamayo/Starship-EVO#2779]
So with the advent of weapons and people placing them on turrets, people soon realised that not only do addons on weapon blocks look ugly, it's also hard to build weapon block/barrels with addons for turrets, without having them hampered immensely.

Therefore, a few boys on the discord have come up with a proposal for weapon building that would embed add-ons within a turret/hull, so that not only can weapon core/barrel setups move on a turret axis without being hampered with the rather large addons, barrels, in general, can look nicer without having to account for the ugly addons

Here is an example of this proposed new weapon build system:
<img width="591" alt="finalweapone" src="https://user-images.githubusercontent.com/43084360/90356334-547d2480-e0a3-11ea-85c2-181eff10a8d8.png">

The weapon core wouldn't stretch, barrels would stretch further, and a new addon rack would be a thing, where you attach addons, and then you use the link tool to link the addon rack to the weapon core (as they are on different entities, but for a static weapon, they don't need to be). It uses all the blocks we have now (none need to be removed), and just requires a new block.

I believe that this or other ways of hiding the addons are important as it enhances turreted weaponry and the aesthetics of regular weapons, making both more viable in the aesthetics department, and making turrets more usable as bulky add-ons won't obstruct their movement.

**Comment by BrickMacklin** [comment]
I like this. I won't call the weapon blocks as they are now straight up _ugly_ but they really do not go with the aesthetic of any of my ships or the ones I see other users making. Mainly because of the lights looking out of place and if you place add-ons then forget it matching the rest of your build. I like the idea of the weapon core and attachments in game as is. But making them hidden so we can have more freedom in the way our barrels look is far preferable if this is something that can be done. It would go a long way for many build styles.

**Comment by Arutus** [comment]
I agree with this idea the amount of aesthetic and functional potential behind turret designing using this is huge

**Comment by tsunamayo** [comment]
Yes I remember the old post, I will also answer there.
So I think it is convoluted way of solving a simple issue.
It would take time to do, add some unnecessary complexity to the game, and add a lot of new issues also, so I am not a big fan of this.
Why dont you guy start by telling me what you dont like with addons? I can tweak them, that seems the simplest way to address that issue.
Also I will likely change the way addon works, I will likely only keep a single shape but it would have different tiers.
Thanks

**Comment by Arutus** [comment]
my current issue with addons is how huge they are and the mount points on barrels, both my and ultras suggestions were just a way to find another method of mounting addons somewhere other than barrels. If the addons could have a wireless dock that you add them to and link to the weapon block it would allow you to basically place addons anywhere aside from on the barrel, which in my opinion is the worst spot to put them

**Comment by ultrak2k** [comment]
they are massive, look bad because they mount on basically what is the barrel, and therefore hamper aesthetics and turret barrel movement

**Comment by Arutus** [comment]
They are also very ugly for turrets as it makes the barrels look rediculous

**Comment by StarrKiss** [comment]
@tsunamayo  They have three problems as far as I can see: 
 

1. They limit the flexibility of turrets and other weaponry. For example, battleship style turrets are essentially impossible without looking terrible and limiting barrel movement, as the addons have to attach to the weapon itself.
2. They limit depth and user optimization. With the limited number of addons along one axis, the user doesn't really need to optimize besides insuring that the turret doesn't overheat. I know that this game is much simpler than things like Starbase or From The Depths, but in THOSE games, their addon system requires the user to essentially solve puzzles as to how to make good weaponry that is focused on a specific task.
3. They limit flexibility in aesthetics. With the current system, there are a whole host of ship designs/aesthetics that simply don't work. Such as: Steampunk ships, (as the barrels require glowy addons, making them impossible to hide effectively). Historical replica ships (see above). More pragmatic/rusty ships (glowy doesn't really fit the aesthetic, ESP. as the actual barrel cannot be painted, only lights.), and, to a lesser extent, large ships: The small caliber size of the weapon system and limited customizability limits the weapon system to mid/sized frigates or smaller fighters, as otherwise, large ships would look ridiculous with many low power weapons (Which, on an unrelated note, is hard to do because it is impossible to copy/paste turrets).

So I think this covers the problems most people have with the current weapon system! I feel like the above proposed system isn't perfect, but is quite a bit better, as the current system limits ship building flexibility.

**Comment by tsunamayo** [comment]
@StarrKiss how is the system working in Starbase?
Also how do you build larger gun in those game, do they have a rescale feature?
Regarding aesthetic I will always run into a gameplay vs style issue.
Also I dont want to make something overcomplicated.

**Comment by StarrKiss** [comment]
@tsunamayo  I can't speak for Starbase as it isn't out, but based on what I've seen and what the devs have said, it's going to work similar to FtD where there is a core element (called the firing core) that you connect- Barrels (of numerous types), loaders, connectors, heat sinks, etc to it. (Of course, in ST:EVO you could remove half of those components to reduce complexity) The turrets are essentially autosetup by placing a turret base, weapon, and vertical rotator on it, and they autolink. Each firing core can have multiple barrels to make battleship style turrets.

In order to make larger guns in those games, you add on a 'caliber increase' module to the turrets and as such can size them from any size to any size. 

FtD/Starbase style weapons cover both good gameplay (in terms of engaging design) and look good. One thing this system allows is the ability to place a turret rotator UNDER the deck and have only the barrels of a turret and the vertical rotator, like so: 
![image](https://user-images.githubusercontent.com/13007697/91626764-a4979900-e977-11ea-99e6-dc2372589ff1.png)
FtD is not the nicest looking game, but in StEVO it could look very nice. 

As for overcomplication- I get it, stevo isn't a realistic sim and its more arcadey building, but I think removing 70% of the elements of FtDs weapons would work well. Say, a system with only four components: Firing Core, Barrel, Loader, and Addons, attached with connectors.

**Comment by tsunamayo** [comment]
They will likely be loader / magazine barrel of some kind for kinetic weapons. You can also place your rotator under the deck in Starship EVO.

**Comment by StarrKiss** [comment]
@tsunamayo  you can, but there is no reason to do so as the weapon is the same as the barrel for energy weapons. And in addition, the weapons are fixed size, so theres not much flexibility with them. As the original user suggested, having a chunk of the weapon rotate independtly of the barrel would lead to much better usability.
